pkg/crc/segment/segment.go

```diff
@@ -140,3 +144,15 @@ func (l *loggingAdapter) Logf(format string, args ...interface{}) {
 func (l *loggingAdapter) Errorf(format string, args ...interface{}) {
 	logging.Errorf(format, args...)
 }
+
+func errorType(err error) string {
+	var preflightError crcErr.PreflightError
+	switch {
+	case errors.Is(err, crcErr.VMNotExist):
+		return "VMNotExit"
+	case errors.Is(err, &preflightError):
+		return "PreflightError"
+	default:
+		return fmt.Sprintf("%T", err)
```

Comment:
Isn't `"%T"` creating strings containing `VMNotExist` or `PreflightError`? A bit surprised they need to be special-cased here.

Reply:
>  is this because they are wrapped inside another error? If yes, maybe we should do some error unwrapping? Or it does not make sense in general?

All the error types are going to be `*errors.SerializableError` ? we can have well known error using the `unwrap` which is just one level below in the error chain.

Reply:
> we can have well known error using the unwrap which is just one level below in the error chain.

Yes, I'd test for this specific `*errors.SerializableError` type and unwrap, hopefully it's enough:

```
var serializableErr *errors.SerializableError
if errors.Is(err, &serializableErr) {
    err = serializableErr.Unwrap()
}
return fmt.Sprintf("%T", err)
```
(totally untested code ^^)

Reply:
@cfergeau we can use unwarp method for any error and it will return `nil` if the `err` doesn't implement it otherwise it will return the unwrapped error.

Reply:
Yup, sounds good as well